File: firefly/src/main/java/com/firefly/server/http2/router/impl/HTTPMethodMatcher.java
package com.firefly.server.http2.router.impl;

/**
 * @author Pengtao Qiu
 */
public class HTTPMethodMatcher extends AbstractPreciseMatcher {

    @Override
    public MatchResult match(String value) {
        return super.match(value.toUpperCase());
    }

    @Override
    public MatchType getMatchType() {
        return MatchType.METHOD;
    }
}

File: firefly-net/src/main/kotlin/com/fireflysource/net/http/server/impl/matcher/HttpMethodMatcher.kt
package com.fireflysource.net.http.server.impl.matcher

import com.fireflysource.net.http.server.Matcher
import com.fireflysource.net.http.server.Router
import java.util.*

class HttpMethodMatcher : AbstractPreciseMatcher() {

    override fun add(rule: String, router: Router) {
        super.add(rule.uppercase(Locale.getDefault()), router)
    }

    override fun match(value: String): Matcher.MatchResult? {
        return if (routersMap.isEmpty()) null else super.match(value.uppercase(Locale.getDefault()))
    }

    override fun getMatchType(): Matcher.MatchType {
        return Matcher.MatchType.METHOD
    }

}